modules/nf-core/propr/propr/templates/propr.R

```diff
@@ -61,6 +61,73 @@ read_delim_flexible <- function(file, header = TRUE, row.names = 1, check.names
     return(mat)
 }
 
+#' @return output Data frame
+read_delim_flexible_raw <- function(file, header = TRUE, row.names = NULL, check.names = TRUE){
+
+    ext <- tolower(tail(strsplit(basename(file), split = "\\\\.")[[1]], 1))
+
+    if (ext == "tsv" || ext == "txt") {
+        separator <- "\\t"
+    } else if (ext == "csv") {
+        separator <- ","
+    } else {
+        stop(paste("Unknown separator for", ext))
+    }
+
+    mat <- read.delim(
+        file,
+        sep = separator,
+        header = header,
+        row.names = row.names,
+        check.names = check.names
+    )
+
+    return(mat)
+}
+
+writeGCTFile <-  function(countFile,gctFile){
+    # Following instructions on .gct format from: https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#TXT:_Text_file_format_for_expression_dataset_.28.2A.txt.29
+    counts <- read_delim_flexible_raw(
+        countFile,
+        header = TRUE,
+        row.names = NULL,
+        check.names = FALSE
+    )
+    mat <- read_delim_flexible(
+        countFile,
+        header = TRUE,
+        row.names = opt\$features_id_col,
+        check.names = FALSE
+    )
+
+    mat = t(mat)
+    dimensions <- dim(mat)
+    sampleN <- dimensions[1]
+    geneN <- dimensions[2]
+
+    fileConn<-file(gctFile)
+    writeLines(c("#1.2",paste(geneN,sampleN,sep='\t')), fileConn)
+    close(fileConn)
+
+    samples <- row.names(mat)
+
+    gctDF <- counts[c('gene_name',samples)]
```

Comment:
'gene_name' and 'gene_id' should not be hard coded.
It is possible that the user give a file where instead of 'gene_id' it says 'ensembl_id' or anything else.

Reply:
for this you can check opt$features_id_col

Reply:
Done, I added an arguement to the template files called features_name_col to handle this.